patternfly/patternfly-elements
# fix: Get nested accordions and tabs working inside the content-set

If an accordion or tab is nested inside a pfe-content-set component, the component should upgrade with no errors.

### Related issues

- (#968) Bug in mobile-view if pfe-content-set contains a nested accordion

### Preview

Link(s) to demo page(s) where this element can be viewed:
- [Content set preview](https://deploy-preview-975--happy-galileo-ea79c4.netlify.app/elements/pfe-content-set/demo/) 


### What has changed and why

Summarize files edited as part of this MR along with a brief description of what was changed/why.

- Added example nested tab and accordion on the pfe-content-set demo page
- Assign a random value to `pfe-id` attribute on the upgraded tab or accordion wrapper
- Remove `on` from the schema
- Add test cases
- Prefix all random IDs with `pfe-` so that they never fail on starting with a number


### Testing instructions

Be sure to include detailed instructions on how your update can be tested by another developer.

1. On the demo page, navigate down to the content-set under the heading: "Accordion nested inside a content-set component"
2. Adjust your window to 600px and refresh.
3. The component should upgrade to accordions.
4. Open the first accordion item.  You should see a nested accordion.
5. Open the second accordion item.  You should see a nested accordion.
6. Increase your browser width to 1200px.  Refresh.
7. The component should upgrade to tabs.
8. Open the third tab item.  You should see a nested tabset.


#### Browser requirements

Your component should work in all of the following environments:



### Ready-for-merge Checklist

Check off items as they are completed.  Feel free to delete items if they are not applicable.


### Merging

Please **squash** when merging and ensure your commit message uses [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/#summary) formatting.

**Be sure to share your updates with the [patternfly-elements-contribute@redhat.com](mailto:patternfly-elements-contribute@redhat.com) mailing list!**
